src/main/java/frc/utils/calibration/sysid/SysIdObject.java

```diff
@@ -0,0 +1,110 @@
+            double rampRateVoltsPerSecond) {
+        this(
+                isCTRE,
+                subsystem,
+                voltageSetControl,
+                voltageStepVolts,
+                rampRateVoltsPerSecond,
+                SysIdConstants.DEFAULT_TIMEOUT_SECONDS
+        );
+    }
+
+    public SysIdObject(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl, double voltageStep) {
+        this(
+                isCTRE,
+                subsystem,
+                voltageSetControl,
+                voltageStep,
+                SysIdConstants.DEFAULT_RAMP_RATE_VOLTS_PER_SECOND,
+                SysIdConstants.DEFAULT_TIMEOUT_SECONDS
+        );
+    }
+
+    public SysIdObject(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl) {
+        this(
+                isCTRE,
+                subsystem,
+                voltageSetControl,
+                SysIdConstants.DEFAULT_VOLTAGE_STEP,
+                SysIdConstants.DEFAULT_RAMP_RATE_VOLTS_PER_SECOND,
+                SysIdConstants.DEFAULT_TIMEOUT_SECONDS
+        );
+    }
+
+    public SysIdObject(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageControl, double voltageStepVolts,
+            double rampRateVoltsPerSecond, double timeoutSeconds) {
+        this.usedSubSystem = subsystem;
+        this.isCTRE = isCTRE;
+
+        final SysIdRoutine.Config config = new SysIdRoutine.Config(
+                Volts.of(rampRateVoltsPerSecond).per(Seconds.of(1)),
+                Volts.of(voltageStepVolts),
+                Seconds.of(timeoutSeconds),
+                this.isCTRE
+                        ? (state) -> SignalLogger.writeString("state", state.toString())
+                        : (state) -> Logger.recordOutput("state", state.toString())
+        );
+        final SysIdRoutine.Mechanism mechanism = new SysIdRoutine.Mechanism(
+                (Measure<Voltage> volts) -> voltageControl.accept(volts.in(Volts)),
+                null,
+                usedSubSystem,
+                usedSubSystem.getName()
+        );
+
+        this.sysIdRoutine = new SysIdRoutine(config, mechanism);
+    }
+
+    public Command getSysIdCommand(boolean isQuasistatic, SysIdRoutine.Direction direction) {
+        return isQuasistatic ? getSysIdQuasistatic(direction) : getSysIdDynamic(direction);
+    }
+
+    public Command getSysIdQuasistatic(SysIdRoutine.Direction direction) {
+        final Command command = sysIdRoutine.quasistatic(direction);
+        return addRequirementsAndReturnCommand(command);
+    }
+
+    public Command getSysIdDynamic(SysIdRoutine.Direction direction) {
+        final Command command = sysIdRoutine.dynamic(direction);
+        return addRequirementsAndReturnCommand(command);
+    }
+
+    private Command addRequirementsAndReturnCommand(Command sysIdCommand) {
+        sysIdCommand.addRequirements(usedSubSystem);
+        return isCTRE ? getCTRECommand(sysIdCommand) : sysIdCommand;
+    }
```

Comment:
shouldn't this be two seperate functions?
`addRequierments` and `getAprropriateCommand`?

Reply:
dont see a reason to split it, i mean i want the ctre command the be alone in function, so add one more function to do the check doesnt feel necessery, but i can do that

Reply:
a function should only have one purpose, split it

Reply:
no, when you split this func you give a option to get a command without add requiremenrts, also this function have one purpose and that to return appropriate command even if i split it this function will do 2 things

Reply:
![image](https://github.com/GreenBlitz/Off-Season-2024/assets/139997639/fd1f6073-848d-479b-8be7-ebf780f9e433)